[New Resources:] `azurerm_cdn_frontdoor` - Support for new CDN Frontdoor (#15140)
hashicorp/terraform-provider-azurerm: website/docs/r/cdn_frontdoor_origin_group.html.markdown

  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_cdn_frontdoor_origin_group" "example" {
  name                         = "example-originGroup"
  cdn_cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id

  health_probe {
    interval_in_seconds = 240
    path                = "/healthProbe"
    protocol            = "Https"
    request_type        = "GET"
  }

  load_balancing {
    additional_latency_in_milliseconds = 0
    sample_size                        = 16
    successful_samples_required        = 3
  }

  session_affinity                      = true
  restore_traffic_or_new_endpoints_time = 10
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Frontdoor Origin Group. Changing this forces a new Frontdoor Origin Group to be created.

* `cdn_frontdoor_profile_id` - (Required) The ID of the Frontdoor Profile. Changing this forces a new Frontdoor Origin Group to be created.

* `load_balancing` - (Required) A `load_balancing` block as defined below.

* `health_probe` - (Optional) A `health_probe` block as defined below.

> * `session_affinity` - (Optional) Whether to allow session affinity on this host. Possible values are `true` or `false`. Defaults to `true`.

Reviewer: ```suggestion
* `session_affinity_enabled` - (Optional) Whether to allow session affinity on this host. Possible values are `true` or `false`. Defaults to `true`.
```

Author: Fixed

> * `restore_traffic_or_new_endpoints_time` - (Optional) Time in minutes to shift the traffic to another endpoint when an healthy endpoint becomes unhealthy or a new endpoint is added. Default is `10` minutes.

Reviewer: we should include the unit in the property name
```suggestion
* `restore_traffic_or_new_endpoints_after_min` - (Optional) Time in minutes to shift the traffic to another endpoint when an healthy endpoint becomes unhealthy or a new endpoint is added. Default is `10` minutes.
```

Author: Fixed

> * `restore_traffic_or_new_endpoints_after_minutes` - (Optional) Time in minutes to shift the traffic to another endpoint when an healthy endpoint becomes unhealthy or a new endpoint is added. Default is `10` minutes.

Reviewer 2: The name of this argument seems strange. "restore traffic **_or_** new endpoints"? The meaning is not clear.

The name of the corresponding property in the REST API is `trafficRestorationTimeToHealedOrNewEndpointsInMinutes`. The "healed or new" part seems important. Maybe the argument name could be improved?

Also the description is wrong:
```suggestion
* `restore_traffic_or_new_endpoints_after_minutes` - (Optional) Time in minutes to shift the traffic to the endpoint gradually when an unhealthy endpoint comes healthy or a new endpoint is added. Default is `10` minutes.
```

Author: I went with `restore_traffic_time_to_healed_or_new_endpoint_in_minutes`, it's a little bit shorter but still is true to the underlying API naming.

Fixed.

~> **NOTE:** This property is currently not used, but will be in the near future.

---

A `health_probe` block supports the following:

* `interval_in_seconds` - (Optional) The number of seconds between health probes. Default is `240` seconds. Possible values are between `5` and `31536000` seconds(inclusive).

* `path` - (Optional) The path relative to the origin that is used to determine the health of the origin. Defaults to `/`.

* `protocol` - (Optional) Protocol to use for health probe. Possible values are `Http`, `Https` or `NotSet`. Defaults to `Https`.

* `request_type` - (Optional) The type of health probe request that is made. Possible values are `GET`, `HEAD` or `NotSet`. Defaults to `GET`.

---

A `load_balancing` block supports the following:

* `additional_latency_in_milliseconds` - (Optional) The additional latency in milliseconds for probes to fall into the lowest latency bucket. Possible values are between `0` and `1000` seconds(inclusive). Defaults to `0`.

> * `sample_size` - (Optional) The number of samples to consider for load balancing decisions. Possible values are between `0` and `255`(inclusive). Defaults to `16`.

Reviewer: sample size indications the "size of the sample" typically? so should this be
```suggestion
* `sample_count` - (Optional) The number of samples to consider for load balancing decisions. Possible values are between `0` and `255`(inclusive). Defaults to `16`.
```
or
```suggestion
* `samples_considered` - (Optional) The number of samples to consider for load balancing decisions. Possible values are between `0` and `255`(inclusive). Defaults to `16`.
```

Author: Fixed... I went with `sample_count` 😃
